NPE in c.r.d.i.qute.psi.utils.AnnotationUtils.isMatchAnnotation

```java
Internal error: java.lang.NullPointerException: Cannot invoke "String.length()" because "suffix" is null

java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "String.length()" because "suffix" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:708)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
	at com.redhat.devtools.intellij.quarkus.lsp4ij.IndexAwareLanguageClient.runTask(IndexAwareLanguageClient.java:101)
	at com.redhat.devtools.intellij.quarkus.lsp4ij.IndexAwareLanguageClient$1.run(IndexAwareLanguageClient.java:76)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:429)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressInCurrentThread$11(CoreProgressManager.java:557)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressInCurrentThread(CoreProgressManager.java:557)
	at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:363)
	at com.redhat.devtools.intellij.quarkus.lsp4ij.IndexAwareLanguageClient.lambda$runAsBackground$0(IndexAwareLanguageClient.java:73)
	at com.redhat.devtools.intellij.quarkus.lsp4ij.IndexAwareLanguageClient.lambda$runAsBackground$1(IndexAwareLanguageClient.java:87)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NullPointerException: Cannot invoke "String.length()" because "suffix" is null
	at java.base/java.lang.String.endsWith(String.java:2313)
	at com.redhat.devtools.intellij.qute.psi.utils.AnnotationUtils.isMatchAnnotation(AnnotationUtils.java:85)
	at com.redhat.devtools.intellij.qute.psi.internal.java.AbstractQuteTemplateLinkCollector.visitClass(AbstractQuteTemplateLinkCollector.java:139)
	at com.intellij.psi.impl.source.PsiClassImpl.accept(PsiClassImpl.java:464)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.acceptChildren(SharedImplUtil.java:185)
	at com.intellij.psi.impl.source.PsiFileImpl.acceptChildren(PsiFileImpl.java:765)
	at com.intellij.psi.JavaRecursiveElementVisitor.visitElement(JavaRecursiveElementVisitor.java:44)
	at com.intellij.psi.PsiElementVisitor.visitFile(PsiElementVisitor.java:51)
	at com.intellij.psi.JavaElementVisitor.visitJavaFile(JavaElementVisitor.java:212)
	at com.intellij.psi.impl.source.PsiJavaFileBaseImpl.accept(PsiJavaFileBaseImpl.java:478)
	at com.redhat.devtools.intellij.qute.psi.internal.java.QuarkusIntegrationForQute.diagnostics(QuarkusIntegrationForQute.java:51)
	at com.redhat.devtools.intellij.qute.psi.QuteSupportForJava.diagnostics(QuteSupportForJava.java:86)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getJavaDiagnostics$12(QuteLanguageClient.java:134)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:891)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:69)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getJavaDiagnostics$13(QuteLanguageClient.java:134)
	at com.redhat.devtools.intellij.quarkus.lsp4ij.IndexAwareLanguageClient.runTask(IndexAwareLanguageClient.java:97)
	... 22 more
```

_Originally posted by @gastaldi in https://github.com/redhat-developer/intellij-quarkus/issues/748#issuecomment-1538856061_

--- a/src/main/java/com/redhat/devtools/intellij/qute/psi/utils/AnnotationUtils.java
+++ b/src/main/java/com/redhat/devtools/intellij/qute/psi/utils/AnnotationUtils.java
@@ -76,6 +76,9 @@ public class AnnotationUtils {
 	 *         false otherwise.
 	 */
 	public static boolean isMatchAnnotation(PsiAnnotation annotation, String annotationName) {
+		if(annotation == null || annotation.getQualifiedName() == null){
+			return false;
+		}
 		// Annotation name is the fully qualified name of the annotation class (ex :
 		// org.eclipse.microprofile.config.inject.ConfigProperties)
 		// - when IAnnotation comes from binary, IAnnotation#getElementName() =

